From b3914bbd0973aa79d91ca8beca49c9fdd89d04d0 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Wed, 06 Nov 2024 14:38:52 +0200
Subject: [PATCH] [SW-207762] board: fs1: create pallad bootloader hex

Generate pldm_bloader_rom_ecc.hex, which is palladium
bootloader, used for running palladium standalone preboot

Change-Id: I872e8c5bfd32546700bcf0fbd221873eff7bdf53
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh b/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
index e900ec2..1c259f3 100755
--- a/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
+++ b/board/habanalabs/fs1_arc/fs1-post-image-spi-arc_global.sh
@@ -6,6 +6,7 @@
 
 SHA256_SIZE=32
 SHA256_CHKSUM=sha256sum
+BOOTLOADER_BINFILE=pldm_bloader.bin
 HL_BASE_PATH=${BR2_EXTERNAL}/board/habanalabs
 BOARD_BASE=${HL_BASE_PATH}/${HL_DEV}
 DEVICE_BASE=${HL_BASE_PATH}/${HL_DEV}
@@ -405,6 +406,9 @@
 		mv boot_rom_comp_*.hex ${HEXS_DIR}
 		mv fs1-pldm-bootrom_comp_*_ecc.hex ${HEXS_DIR}
 
+		# convert io die bootloader to pallad hex image
+		${SCRIPTS_DIR}/palad/convert_app_to_io_bootrom_hex.sh ${BOOTLOADER_BINFILE}
+
 		# convert pif.arc application to pallad hex images
 		${SCRIPTS_DIR}/palad/convert_app_to_ccm_hex.sh ${ZEPHYR_DSM_BINFILE}
 		mv iccm0_0.hex ${HEXS_DIR}/pif_iccm0_0.hex
